test/CuberilleTest04.cxx

```diff
@@ -0,0 +1,97 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include <itkTestingMacros.h>
+#include <itkImage.h>
+#include <itkConstantPadImageFilter.h>
+#include <itkQuadEdgeMesh.h>
+#include <itkCuberilleImageToMeshFilter.h>
+
+// In the case where a 2x2x2 region contains two background pixels at
+// opposite corners, and foreground pixels elsewhere, it is necessary
+// that there be two vertices at the center.  Previously, this was
+// handled incorrectly with a single vertex, causing non-manifold
+// geometry.  This test covers the four cases in which this occurs.
+
+int
+CuberilleTest04(int itkNotUsed(argc), char * itkNotUsed(argv)[])
+{
+
+  using TPixel = unsigned char;
+  using TCoordinate = double;
+  const unsigned int Dimension = 3;
+  using TImage = itk::Image<TPixel, Dimension>;
+  using TMesh = itk::QuadEdgeMesh<TCoordinate, Dimension>;
+  using TPad = itk::ConstantPadImageFilter<TImage, TImage>;
+  using TExtract = itk::CuberilleImageToMeshFilter<TImage, TMesh>;
+
+  // 126 01111110
+  // 189 10111101
+  // 219 11011011
+  // 231 11100111
+
+  std::array<size_t, 4> masks{ 126, 189, 219, 231 };
+
+  for (const auto & mask : masks)
+  {
+
+    std::bitset<8> bitmask(mask);
+
+    std::cout << mask << ' ' << bitmask << std::endl;
+
+    const auto       image = TImage::New();
+    TImage::SizeType size;
+    size.Fill(2);
+
+    TImage::IndexType origin;
+    origin.Fill(0);
+
+    TImage::RegionType region(origin, size);
+
+    image->SetRegions(region);
+    image->Allocate();
+    image->FillBuffer(0);
+
+    for (size_t index = 0; index < pow(2, 3); ++index)
```

Comment:
Not sure if this calls `std::pow` or else a power function exists within ITK. I've seen that `VNL` or `Eigen` also define it. In any case, if this calls to the std library power function, if I'm not mistaken, we prefer use the namespace as well to make it clear the function we call.

And although it will not case much of a difference in terms of performance, or it maybe more of a personal choice, maybe we could have it pre-computed in a variable.

Reply:
Done in 81359c5. Thanks @DVigneault  !